release-notes/8.0/preview/api-diff/preview4/Microsoft.NETCore.App/8.0-preview4_System.IO.md

```diff
@@ -0,0 +1,43 @@
+# System.IO
+
+``` diff
+ namespace System.IO {
+     public class DirectoryNotFoundException : IOException {
```

Comment:
these ctors got obsoleted, not added: https://github.com/dotnet/runtime/pull/84383

Reply:
It looks better now, but I wonder if it could be improved: no `+` or `-` for the `ctor` itself, just mention the ctor with no prefx (no diff), and two new lines (`+`) for the attributes?

```diff
+        [EditorBrowsableAttribute(EditorBrowsableState.Never)]
+        [ObsoleteAttribute("This API supports obsolete formatter-based serialization. It should not be called or extended by application code.", DiagnosticId="SYSLIB0051", UrlFormat="https://aka.ms/dotnet-warnings/{0}")]
         protected DirectoryNotFoundException(SerializationInfo info, StreamingContext context);
```

Reply:
I agree it would be nice to show such diffs as you describe. However, adding changes to the AsmDiff tool is time consuming and I currently cannot address it. If you don't mind, I'll merge this PR with the addressed fixes and will open an issue in the arcade repo to consider your suggestion.

Reply:
https://github.com/dotnet/arcade/issues/13814